test/test_mahalanobis_mixin.py

```diff
@@ -0,0 +1,203 @@
+from itertools import product
+
+import pytest
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+from sklearn import clone
+from sklearn.datasets import load_iris
+from sklearn.utils import check_random_state, shuffle
+
+from metric_learn import (Constraints, ITML, LSML, MMC, SDML, Covariance, LFDA,
+                          LMNN, MLKR, NCA, RCA)
+from metric_learn.constraints import wrap_pairs
+from functools import partial
+
+
+def build_data():
+  RNG = check_random_state(0)
+  dataset = load_iris()
+  X, y = shuffle(dataset.data, dataset.target, random_state=RNG)
+  num_constraints = 20
+  constraints = Constraints.random_subset(y)
+  pairs = constraints.positive_negative_pairs(num_constraints,
+                                              same_length=True,
+                                              random_state=RNG)
+  return X, pairs
+
+
+def build_pairs():
+  # test that you can do cross validation on tuples of points with
+  #  a WeaklySupervisedMetricLearner
+  X, pairs = build_data()
+  pairs, y = wrap_pairs(X, pairs)
+  pairs, y = shuffle(pairs, y)
+  return pairs, y
+
+
+def build_quadruplets():
+  # test that you can do cross validation on a tuples of points with
+  #  a WeaklySupervisedMetricLearner
+  X, pairs = build_data()
+  c = np.column_stack(pairs)
+  quadruplets = X[c]
+  quadruplets = shuffle(quadruplets)
+  return quadruplets, None
+
+
+list_estimators = [(Covariance(), build_data),
+                   (ITML(), build_pairs),
+                   (LFDA(), partial(load_iris, return_X_y=True)),
+                   (LMNN(), partial(load_iris, return_X_y=True)),
+                   (LSML(), build_quadruplets),
+                   (MLKR(), partial(load_iris, return_X_y=True)),
+                   (MMC(), build_pairs),
+                   (NCA(), partial(load_iris, return_X_y=True)),
+                   (RCA(), partial(load_iris, return_X_y=True)),
+                   (SDML(), build_pairs)
+                   ]
+
+ids_estimators = ['covariance',
+                  'itml',
+                  'lfda',
+                  'lmnn',
+                  'lsml',
+                  'mlkr',
+                  'mmc',
+                  'nca',
+                  'rca',
+                  'sdml',
+                  ]
+
+
+@pytest.mark.parametrize('estimator, build_dataset', list_estimators,
+                         ids=ids_estimators)
+def test_score_pairs_pairwise(estimator, build_dataset):
+  # Computing pairwise scores should return an euclidean distance matrix.
+  inputs, labels = build_dataset()
+  X, _ = load_iris(return_X_y=True)
+  n_samples = 20
+  X = X[:n_samples]
+  model = clone(estimator)
+  model.fit(inputs, labels)
+
+  pairwise = model.score_pairs(np.array(list(product(X, X))))\
+      .reshape(n_samples, n_samples)
+
+  check_is_distance_matrix(pairwise)
```

Comment:
I think `score_pairs` returns squared distances, no? so this should be called on the element-wise square root of `pairwise`?

Reply:
For now `score_pairs` returns not squared distances, as implemented in `MahalanobisMixin`
https://github.com/wdevazelhes/metric-learn/blob/eff278e007dded4a6f5ee2ead3ae92bb8c720eb6/metric_learn/base_metric.py#L97
Should it return squared distances though ? I will add the TODO to at least give the option because it would prevent some useless computations if the true distance is not needed